src/Symfony/Component/Messenger/Bridge/Amqp/Tests/Transport/ConnectionTest.php

```diff
@@ -285,7 +285,7 @@ public function testItSetupsTheConnectionWithDefaults()
             $amqpExchange = $this->createMock(\AMQPExchange::class)
         );
 
-        $amqpExchange->expects($this->once())->method('declareExchange');
+        $amqpExchange->expects($this->exactly(2))->method('declareExchange');
```

Comment:
Would you explain, why this method call two times?
It would be nice to add an explanation in the code.

Reply:
because the `setup` method declares 2 exchanges

Reply:
I saw that the `delayExchage` declares alwas now, even it was not accessed.
maybe it would be better to declare it here https://github.com/symfony/symfony/blob/0db080e9fcdd2e021846280071ef29aa3cbaa372/src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php#L364

Reply:
This won't work, at this point you wan't know if user disabled `auto_setup`.

I've update the PR to use 2 flags that distinguish autosetup for standard exchange and delayed exchange